File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
import java.util.*;

class Solution {
    
    
    public String[] convertN(int num, int n){
        
        if(num == 0){
            return new String[]{"0"};
        }
        
        StringBuilder result = new StringBuilder("");
        
        String[] alps = new String[]{"A","B","C","D","E","F"};
        
        while (num !=0){
        
            String r = (num % n <=9) ? String.valueOf(num%n) : alps[ (num%n) - 10];
            result.append(r);
            num = (int)(num / n);
        }
        
        //뒤집기
        String answer = result.reverse().toString();
        return answer.split("");
        
    }
    
    public String solution(int n, int t, int m, int p) {
        String answer = "";
        
        
        // 현재 몇번째 값인지 체크하기 위해 m과 p를 사용. m으로 count를 나눈 나머지 -> p-1일때 t-1 -> t가 0이면 그만
        // stop 될때까지 n진수로 변환하여 for문으로 값을 answer에 더해주기
        
        
        int current10Number = 0; // 십진수
        int totalCount = 0;
        
        
        
        while(true){
            
        
            String[] nNumbers = convertN(current10Number,n);
            
            for(int i=0; i<nNumbers.length; i++){
                
                if((totalCount % m) == p-1){
                    answer += nNumbers[i];
                    t-=1;
                    if(t==0){
                        return answer;
                    }
                }
                
                totalCount+=1;
            }
            
            current10Number+=1;
            
        }
    }
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def to_n_number(num, n):
    
    answer = []
    

    if num==0:
        return [0]

    alp = ['A','B','C','D','E','F']
    
    while num!=0:
        
        r = num%n if num%n <=9 else alp[(num%n)-10]
        answer.append(r)
        num = num//n
    
    return answer[::-1]
    
    

def solution(n, t, m, p):
    answer = ''
    
    
    # 게임에 참가하는 인원 m과 튜브의 순서 p가 주어짐 ->  m으로 나눴을 때 나머지가 p인 것들을 출력
    
    # t개를 출력할 때까지 진법 변환을 하고 t개 출력하고 break
    
    current_t = 0
    current_num = 0
    count = 0
    flag = False

    while True:

        if current_t >= t:
            break
            
        n_num_list = to_n_number(current_num, n)
        
        for i in range(len(n_num_list)):
            if (count % m) ==p-1:# m으로 나눴을 때 현재 숫자가 p번째 순서인 경우
                answer+=str(n_num_list[i])
                current_t +=1
                if current_t == t:
                    flag = True
                    break
            
            count+=1
        current_num+=1

        if flag:
            break

            
            
    return answer